index.js

```diff
@@ -160,10 +165,20 @@ Filter.prototype.build = function() {
             return fs.unlinkSync(outputPath);
           } case 'change': {
             instrumentation.change++;
-            return plugin._handleFile(relativePath, srcDir, destDir, entry, outputFilePath, true, instrumentation);
+            var changeOperation = plugin._handleFile(relativePath, srcDir, destDir, entry, outputFilePath, true, instrumentation);
+            if (plugin.async) {
```

Comment:
I think the throttling should happen in the individual plugins that extend this filter, based on the characteristics of that particular plugin - how it utilizes memory, CPU, disk, etc. I don't know if there is a general way to do throttling here that would work well for everything.

For instance since `broccoli-babel-transpiler` is going to use a worker pool, I think it makes sense to handle the throttling there since it has control over that queue length and can be tuned to get the best balance of speed and memory for its use case.